### PR TITLE
[v12] Add rate limiting to Assist

### PR DESCRIPTION
**Pull request by jakule**
Backport of #26011

### CODE REVIEW

**Review by justinas — APPROVED** [review]

File: lib/web/apiserver.go [hunk 1]
```diff
@@ -110,7 +120,14 @@ type Handler struct {
 	sessionStreamPollPeriod time.Duration
 	clock                   clockwork.Clock
 	limiter                 *limiter.RateLimiter
-	healthCheckAppServer    healthCheckAppServerFunc
+	highLimiter             *limiter.RateLimiter
```
Comment (justinas) [inline lib/web/apiserver.go:123]: This is unrelated to the assistant limiter and should not be backported.

Comment (jakule) [reply lib/web/apiserver.go]: Removed

**Review by tigrato — APPROVED** [review]

**Review by codingllama — APPROVED** [review]
